packages/browser-wallet/src/popup/pages/Account/SendCcd/PickToken.tsx

```diff
@@ -0,0 +1,62 @@
+import { AccountTokens, contractBalancesFamily } from '@popup/store/token';
+import { TokenIdAndMetadata } from '@shared/storage/types';
+import { CCD_METADATA, ContractBalances, TokenIdentifier } from '@shared/utils/token-helpers';
+import { atom, Atom } from 'jotai';
+import { selectAtom } from 'jotai/utils';
+import React, { Suspense } from 'react';
+import DisplayToken from './DisplayToken';
+
+interface Props {
+    onClick: (token: TokenIdentifier | undefined) => void;
+    tokens?: AccountTokens;
+    ccdBalance: bigint;
+    address: string;
+}
+
+interface CollectionProps extends Pick<Props, 'onClick'> {
+    balanceAtom: Atom<ContractBalances>;
+    tokens: TokenIdAndMetadata[];
+    contractIndex: string;
+}
+
+function Collection({ balanceAtom, onClick, tokens, contractIndex }: CollectionProps) {
+    return (
+        <>
+            {tokens.map((token) => (
+                <DisplayToken
+                    className="create-transfer__pick-token-element"
+                    key={token.id}
+                    metadata={token.metadata}
+                    balanceAtom={selectAtom(balanceAtom, (balances) => balances[token.id])}
+                    onClick={() => onClick({ contractIndex, tokenId: token.id, metadata: token.metadata })}
+                />
+            ))}
+        </>
+    );
+}
+
+export default function PickToken({ onClick, tokens, ccdBalance, address }: Props) {
```

Comment:
Would be nice if this auto-shrunk the account balance card if >4 tokens are available.